packages/react-meteor-accounts/react-accounts.tsx

```diff
@@ -0,0 +1,160 @@
+    (props: Omit<P, keyof WithUserIdProps> & Partial<WithUserIdProps>, ref) => {
+      const userId = useUserId();
+      return (
+        <Component
+          // Cast because `P` may not include `userId`. TS forces cast to unknown first.
+          {...({ userId } as unknown as P)}
+          ref={ref}
+          {...props}
+        />
+      );
+    }
+  );
+}
+
+/**
+ * Hook to get a stateful value of the current user record. Uses `Meteor.user`, a reactive data source.
+ * @see https://docs.meteor.com/api/accounts.html#Meteor-user
+ */
+export function useUser() {
+  const [user, setUser] = useState(Meteor.user());
+  useEffect(() => {
+    const computation = Tracker.autorun(() => {
+      let user = Meteor.user();
+      // `Meteor.user` returns `undefined` after logout, but that ruins type signature and test parity. So, cast until that's fixed.
+      if (user === undefined) {
+        user = null;
+      }
+      setUser(user);
+    });
+    return () => {
+      computation.stop();
+    };
+  }, []);
+  return user;
+}
+
+export interface WithUserProps {
+  user: Meteor.User | null;
+}
+
+/**
+ * HOC to get a stateful value of the current user record. Uses `Meteor.user`, a reactive data source.
+ * @see https://docs.meteor.com/api/accounts.html#Meteor-user
+ */
+export function withUser<P>(Component: React.ComponentType<P>) {
+  return forwardRef(
+    (props: Omit<P, keyof WithUserProps> & Partial<WithUserProps>, ref) => {
+      const user = useUser();
+      return <Component {...({ user } as unknown as P)} ref={ref} {...props} />;
+    }
+  );
+}
+
+/**
+ * Hook to get a stateful value of whether a login method (e.g. `loginWith<Service>`) is currently in progress. Uses `Meteor.loggingIn`, a reactive data source.
+ * @see https://docs.meteor.com/api/accounts.html#Meteor-loggingIn
+ */
+export function useLoggingIn(): boolean {
+  const [loggingIn, setLoggingIn] = useState(Meteor.loggingIn());
+  useEffect(() => {
+    const computation = Tracker.autorun(() => {
+      setLoggingIn(Meteor.loggingIn());
+    });
+    return () => {
+      computation.stop();
+    };
+  }, []);
+  return loggingIn;
+}
+
+export interface WithLoggingInProps {
+  loggingIn: boolean;
+}
+
+/**
+ * HOC to forward a stateful value of whether a login method (e.g. `loginWith<Service>`) is currently in progress. Uses `Meteor.loggingIn`, a reactive data source.
+ * @see https://docs.meteor.com/api/accounts.html#Meteor-loggingIn
+ */
+export function withLoggingIn<P>(Component: React.ComponentType<P>) {
+  return forwardRef(
+    (
+      props: Omit<P, keyof WithLoggingInProps> & Partial<WithLoggingInProps>,
+      ref
+    ) => {
+      const loggingIn = useLoggingIn();
+      return (
+        <Component {...({ loggingIn } as unknown as P)} ref={ref} {...props} />
+      );
+    }
+  );
+}
+
+/**
+ * Hook to get a stateful value of whether the logout method is currently in progress. Uses `Meteor.loggingOut`, a reactive data source.
+ */
```

Comment:
```suggestion
 * @see https://docs.meteor.com/api/accounts.html#Meteor-loggingOut
 */
```
documentation has since been added.